### PR TITLE
🚑️ fixes issue with env vars being stored as dict or list

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Fixes an issue where env vars stored as dictionaries are not handled by the compose spec validation. This happens specifically for the edge proxies. That's why we only see the issue on deployments with blocked internet access.

- added regression test
- fixed issue with `cancel_sequential_workers`

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by YuryHrytsuk — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: packages/service-library/src/servicelib/async_utils.py [hunk 1]
```diff
@@ -42,14 +42,24 @@ class QueueElement:
...
-    for context in _sequential_jobs_contexts.values():
+async def _safe_cancel(context: Context) -> None:
+    try:
         await context.in_queue.put(None)
         if context.task is not None:
             context.task.cancel()
             with suppress(asyncio.CancelledError):
                 await context.task
+    except RuntimeError as e:
+        if "Event loop is closed" in f"{e}":
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/async_utils.py:53]: Note that you are suppressing a very specific exceptions. Is there another **more reliable** way to filter these exceptions? Where is this coming from?

Some suggestions:
- Is it "normal" that the event loop closes? In general I think this is an anomaly that should be diagnosed and corrected. Therefore, I think this is a warning not an info. 
- If you agree with the latter, I would add more info to debug it e.g. trace and exception

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py [hunk 2]
```diff
@@ -107,14 +107,21 @@ def _apply_templating_directives(
...
     def _gen_parts_env_vars(
-        env_vars: list[str],
+        env_vars: list[str] | dict[str, str],
     ) -> Generator[tuple[str, str], None, None]:
-        for env_var in env_vars:
-            key, value = env_var.split("=")
-            yield key, value
+        assert isinstance(env_vars, list | dict)  # nosec
+
+        if isinstance(env_vars, list):
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py:118]: THOUGHT: This is the reason why we impose certain standards. The env section of our docker-compose are required to be key-value. When you establish certain guarantees at the source, there is no need to be doing these checks. It is better for the Umwelt ;-)

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py [hunk 3]
```diff
@@ -107,14 +107,21 @@ def _apply_templating_directives(
...
+            for env_var in env_vars:
+                key, value = env_var.split("=")
+                yield key, value
+        else:
+            for key, value in env_vars.items():
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/validation.py:123]: TIP: you can yield on an iterator in a one liner as 
```python
yield from env_vars.items()
```

**Review by pcrespov — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]

**Pull request merged**